### DISCUSSION TITLE
[filesystem] collapse_directory

### DISCUSSION THREAD

**Discussion by MunifTanjim** [nvim-neo-tree/neo-tree.nvim#258] [Q&A] [answered]
How do I do this?

```lua
  collapse_directory = function(state)
    local node = state.tree:get_node()
    if node.type ~= "directory" or not node:is_expanded() then
      node = state.tree:get_node(node:get_parent_id())
    end
    if node.type == "directory" then
      -- collapse node
      -- focus node
    end
  end,
```

I want to collapse a directory and then focus it.

**Comment by cseickel** [comment]
The collapse is just the standard NuiNode collapse followed by a NuiTree render. The focus node method is a custom Neo-tree function:

```lua
  collapse_directory = function(state)
    local node = state.tree:get_node()
    if node.type ~= "directory" or not node:is_expanded() then
      node = state.tree:get_node(node:get_parent_id())
    end
    if node.type == "directory" then
      -- collapse node
      node:collapse()
      state.tree:render()
      -- focus node
      require("neo-tree.ui.renderer").focus_node(state, node:get_id())
    end
  end,
```

**Comment by cseickel** [comment, marked as answer]
What I should have said was that this functionality already exists, it is the `"close_node"` command, mapped to `C` by default. That built-in function also works with nested files.

**Reply by gegoune** [reply]
`close_node` will close node when cursor is on file within that node. `toggle_node` has to be on directory itself. Could `toggle_node` also toggle node if cursor is on file within that node?

**Reply by cseickel** [reply]
Moved this to a new discussion: https://github.com/nvim-neo-tree/neo-tree.nvim/issues/260
